src/log.h

```diff
@@ -0,0 +1,89 @@
+#ifndef VALKEY_LOG_H
+#define VALKEY_LOG_H
+
+#include <ctype.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <sys/syslog.h>
+#include <sys/time.h>
+#include <time.h>
+#include <unistd.h>
+
+#define LOG_MAX_LEN 1024 /* Default maximum length of syslog messages.*/
+
+/* Log levels */
+#define LL_DEBUG 0
+#define LL_VERBOSE 1
+#define LL_NOTICE 2
+#define LL_WARNING 3
+#define LL_NOTHING 4
+#define LL_RAW (1 << 10) /* Modifier to log without timestamp */
+
+/* Use macro for checking log level to avoid evaluating arguments in cases log
+ * should be ignored due to low level. */
+#define serverLog(level, ...)                                                                                          \
+    do {                                                                                                               \
+        if (((level) & 0xff) < server.verbosity) break;                                                                \
+        valkeyLog(level, server.syslog_enabled, server.timezone, server.daylight_active, server.sentinel_mode,         \
+                  server.pid, server.primary_host, server.logfile, __VA_ARGS__);                                       \
+    } while (0)
+
+#define serverLogRaw(level, ...)                                                                                       \
+    do {                                                                                                               \
+        if (((level) & 0xff) < server.verbosity) break;                                                                \
```

Comment:
```suggestion
        if (((level) & 0xff) < server.verbosity) break;                                                                \
```
This feels a bit like fake decoupling, since you can't import log.h *without* server.h, since we are accessing the server structure here.

Reply:
Yeah, the existing log implementation seems relied server and server config too much, its not straightforward to split it from `server.h`. Let's open an issue to discuss the code refactor work.